### PR TITLE
Fix app_data access, which I broke in the Actix Web 4 upgrade.

### PR DESCRIPTION
**Pull request by smarnach**
Accessing application state changed in Actix Web 4, and I implemented the changes in #123 only incompletely. With this change, we always use the `Data` wrapper for the endpoint state, which fixes the problem.

### CODE REVIEW

**Review by leplatrem — APPROVED** [review]
Looks good from here.

Should we consider having a type alias for `web::Data<EndpointState>` if they're always used together?

**Comment by smarnach** [comment]
Thanks for the review!

> Should we consider having a type alias for `web::Data<EndpointState>` if they're always used together?

There are a few options to make this nicer. The only purpose of the `Data` wrapper is that it implements the `FromRequest` trait, so we can extract the state via an "extractor" parameter to request handlers. I was slightly confused by the migration guide in this point – I thought this change was kind of required, but I could instead have manually extracted the state from the request object everywhere, so we wouldn't have needed the `Data` wrapper in the first place.

Another option would be to implement `FromRequest` for `EndpointState` directly – the implementation would be trivial, and we also wouldn't need the `Data` wrapper.

Using a type alias might also help, but then we would probably also need to introduce some `new` method for `EndpointState` that returns `Data<EndpointState>`, which I'm not sure is worth it, given that this might be the last update to this code before it gets retired.

If this codebase were to live on, I would factor most of the code in `main()` out into a separate function that could be tested in an integration test that basically runs the whole app. Currently there are only unit tests, which is why the issue we saw in stage was not caught by the tests at all.

In summary, I'd just like to get this running again, and resolve three years worth of security advisories in production.
